lib/engine/game/g_1850/game.rb

```diff
@@ -331,9 +331,40 @@ def revenue_for(route, stops)
         end
 
         def sell_shares_and_change_price(bundle, allow_president_change: true, swap: nil, movement: nil)
-          @sell_queue << [bundle, bundle.corporation.owner]
+          bundle_owner = bundle.owner
+          @sell_queue << [bundle, bundle.corporation.owner] unless bundle.corporation.president?(bundle_owner)
 
           @share_pool.sell_shares(bundle)
+
+          change_price(bundle, bundle.corporation.owner) if bundle.corporation.president?(bundle_owner)
+        end
+
+        def change_price(bundle, corporation_owner, forced = false)
+          corporation = bundle.corporation
```

Comment:
this code is taken from 1870 price_protect class made into a method and moved to the games class